node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -335,32 +331,71 @@ object WebApi {
   private def exprFromBundleProto(b: Bundle): Option[RhoExpr] = exprFromParProto(b.body)
 
   private def mkUnforgExpr(f: String => RhoUnforg, bs: ByteString): ExprUnforg =
-    ExprUnforg(f(toHex(bs)))
+    ExprUnforg(f(bs.toHexString))
 
   // RhoExpr to protobuf
 
-  private def unforgToUnforgProto(unforg: RhoUnforg): GUnforgeable.UnfInstance = unforg match {
-    case UnforgPrivate(name)  => GPrivateBody(GPrivate(name.unsafeHexToByteString))
-    case UnforgDeploy(name)   => GDeployIdBody(GDeployId(name.unsafeHexToByteString))
-    case UnforgDeployer(name) => GDeployerIdBody(GDeployerId(name.unsafeHexToByteString))
+  /**
+    * Private definition of [[Monoid]] for [[Par]] with the implementation of
+    * combining fields used by WebApi representation of Par [[RhoExpr]].
+    */
+  implicit private object ParMonoid extends Monoid[Par] {
+    override def empty: Par = VectorPar()
+
+    override def combine(x: Par, y: Par): Par =
+      x.copy(
+        x.sends ++ y.sends,
+        x.receives ++ y.receives,
+        x.news ++ y.news,
+        x.exprs ++ y.exprs,
+        x.matches ++ y.matches,
+        x.unforgeables ++ y.unforgeables,
+        x.bundles ++ y.bundles,
+        x.connectives ++ y.connectives
+      )
+  }
```

Comment:
Is it simpler than just `foldLeft`? I guess it has the same complexity but has more specific implementation.

Reply:
The main advantage is to be compositional e.g. to get `Monoid[List[Int]]` we can _reuse_ `Monoid[Int]`.
```scala
import cats.syntax.all._

val r2 = 4 |+| 2 // 6

val r1 = List(4) |+| List(2) // List(4, 2)
```
Here is a very good article with concise explanation what problems can solve Semigroup and Monoid.
https://blog.rockthejvm.com/semigroups-and-monoids-in-scala/